src/v/cluster/controller_backend.cc

```diff
@@ -386,13 +387,26 @@ ss::future<> controller_backend::reconcile_ntp(deltas_t& deltas) {
                                  && d.new_assignment.replicas
                                       == it->new_assignment.replicas;
                       });
-                    vassert(
-                      fit == std::next(it),
-                      "finish operation command, if present have to be the one "
-                      "that follows update operation");
 
+                    // The only delta type permitted between `update` and
+                    // `update_finished` is `update_properties`.  Apply any
+                    // intervening deltas of this type and skip
+                    // the cursor ahead to the `update_finished`
                     if (fit != deltas.end()) {
```

Comment:
I don't think so, because if `fit == deltas.end()` then it's permitted for `std::next(it) != end` -- this is what we would see if there was an update, then an update_properties, but no subsequent update_finished yet.

Reply:
ah, of course!